### PR TITLE
TST: fix casting extension tests for external users

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
cc @simonjayhawkins fixtures defined by pandas are not known when inheriting the tests as external package. There are a bunch of fixtures that external packages are expected to define in `pandas/tests/extension/conftest.py`, but I think for the string dtype here this is not worth it to expect downstream users to define it themselves (certainly given that it is still going to change)

### CODE REVIEW

File: pandas/tests/extension/base/casting.py [hunk 1]
```diff
@@ -43,8 +45,19 @@ def test_astype_str(self, data):
...
+        [
+            "string",
+            pytest.param(
+                "arrow_string", marks=td.skip_if_no("pyarrow", min_version="1.0.0")
+            ),
+        ],
+    )
     def test_astype_string(self, data, nullable_string_dtype):
         # GH-33465
+        from pandas.core.arrays.string_arrow import ArrowStringDtype  # noqa: F401
```
Comment (jreback) [inline pandas/tests/extension/base/casting.py:59]: why is this needed?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/extension/base/casting.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/tests/extension/base/casting.py:59]: To force register the dtype, I think (the same reason that @simonjayhawkins also has been adding it in several places). AFAIU, once the ArrowStringDtype is merged with StringDtype, that will be done automatically and all those imports can be removed.

Comment (simonjayhawkins) [reply pandas/tests/extension/base/casting.py:59]: yep. these are removed in #39908

**Review by simonjayhawkins — APPROVED** [review]
Thanks @jorisvandenbossche lgtm.

**Comment by jorisvandenbossche** [comment]
Failure is the flaky ResourceWarning

File: pandas/tests/extension/base/casting.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/extension/base/casting.py:59]: why can't you do the import IN the fixture itself? having in the test is very odd (when you *also* have the fixture)

Comment (jorisvandenbossche) [reply pandas/tests/extension/base/casting.py:59]: There is no fixture to import it inside, only a `pytest.mark.parametrize` decorator

Comment (jreback) [reply pandas/tests/extension/base/casting.py:59]: the `nullable_string_dtype` fixtures, sure there is

Comment (jreback) [reply pandas/tests/extension/base/casting.py:59]: we do this type of thing in the parquet tests, need to do the same here

Comment (jorisvandenbossche) [reply pandas/tests/extension/base/casting.py:59]: Can you be more specific as "this type of thing in the parquet tests" ?

> the nullable_string_dtype fixtures, sure there is

The whole point of this PR is to not use that, please look at the diff. I am removing the `nullable_string_dtype` fixture with a local parametrization

Comment (jreback) [reply pandas/tests/extension/base/casting.py:59]: why, this is *exactly* the reason this fixture should exist. to avoid having to do a local import like this. which will surely be done incorrectly the next time. a fixture exists for exactly this reason.

Comment (jorisvandenbossche) [reply pandas/tests/extension/base/casting.py:59]: The local imports are only temporary, they will be removed in https://github.com/pandas-dev/pandas/pull/39908

Comment (jreback) [reply pandas/tests/extension/base/casting.py:59]: ahh ok great

**Review by jreback — APPROVED** [review]
